library/ssl_tls.c

```diff
@@ -1000,56 +1005,52 @@ static int ssl_tls12_populate_transform( mbedtls_ssl_transform *transform,
     }
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-    ret = mbedtls_cipher_setup_psa( &transform->cipher_ctx_enc,
-                                    cipher_info, transform->taglen );
-    if( ret != 0 && ret != MBEDTLS_ERR_CIPHER_FEATURE_UNAVAILABLE )
-    {
-        MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_cipher_setup_psa", ret );
+    if( ( status = mbedtls_ssl_cipher_to_psa( cipher_info->type,
+                                 transform->taglen,
+                                 &alg,
+                                 &key_type,
+                                 &key_bits ) ) != PSA_SUCCESS )
+    {
+        ret = psa_status_to_mbedtls( status );
+        MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_cipher_to_psa", ret );
         goto end;
     }
 
-    if( ret == 0 )
-    {
-        MBEDTLS_SSL_DEBUG_MSG( 3, ( "Successfully setup PSA-based encryption cipher context" ) );
```

Comment:
Ok, some tests in `ssl-opt.sh` were looking for that pattern. It made sense at the time, because in addition to the compile-time flag, there was also some runtime logic to attempt to use PSA and fallback to non-PSA otherwise - this is a remnant from the time PSA didn't support everything TLS needed. Now that the selection happens purely at compile-time (we no longer even have a `cipher_ctx` when `USE_PSA_CRYPTO` is enabled), I don't think there's any need to check for this in tests.

So, I suggest simply removing those assertions from the tests in `ssl-opt.sh` that have them: `run_test_psa()` and `run_test_psa_force_curve()`. We can also remove the related negative assertions `-C/-S "Failed to setup PSA-based cipher context"`.